File: php/testcode/hardcodedcreds_026.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_hardcoded_encryption_key
function hardcodedcreds026(BenchmarkRequest $req): BenchmarkResponse {
    define('ENC_KEY', 'key1234567890abcd'); // vuln-code-snippet vuln-line php_hardcoded_encryption_key
    $data = $req->param('data');
    $iv = random_bytes(16);
    $ciphertext = openssl_encrypt($data, 'AES-128-CBC', ENC_KEY, 0, $iv);
    return BenchmarkResponse::ok(base64_encode($iv) . ':' . $ciphertext);
}
// vuln-code-snippet end php_hardcoded_encryption_key

File: rust/testcode/hardcodedcreds_026.rs
//! CWE-798: Hardcoded SMTP password stored in a const used in connection setup.

// vuln-code-snippet start testcodeHardcodedcreds026
const SMTP_PASSWORD: &str = "Smtp@P4ssw0rd2024"; // vuln-code-snippet target-line testcodeHardcodedcreds026

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let recipient = req.param("recipient");
    let result = format!(
        "Connecting to smtp.example.com with password={} to send mail to {}",
        SMTP_PASSWORD, recipient
    );
    super::shared::BenchmarkResponse::ok(&result)
}
// vuln-code-snippet end testcodeHardcodedcreds026
